src/cpp/cpp_typecheck_expr.cpp

```diff
@@ -2577,28 +2577,30 @@ void cpp_typecheckt::typecheck_function_call_arguments(
     }
   }
 
-  for(std::size_t i=0; i<parameters.size(); i++)
+  exprt::operandst::iterator arg_it=expr.arguments().begin();
+  for(const auto &parameter : parameters)
   {
-    if(parameters[i].get_bool("#call_by_value"))
+    if(parameter.get_bool("#call_by_value"))
     {
-      assert(is_reference(parameters[i].type()));
+      assert(is_reference(parameter.type()));
 
-      if(expr.arguments()[i].id()!=ID_temporary_object)
+      if(arg_it->id()!=ID_temporary_object)
       {
         // create a temporary for the parameter
 
         exprt arg("already_typechecked");
-        arg.copy_to_operands(expr.arguments()[i]);
+        arg.copy_to_operands(*arg_it);
 
         exprt temporary;
-        new_temporary(expr.arguments()[i].source_location(),
-                      parameters[i].type().subtype(),
+        new_temporary(arg_it->source_location(),
```

Comment:
The parameter should be on the next line. It might be worth rebasing again so you get the lint output from just modified lines.